app/Http/Controllers/Sourcing/PrfController.php

```
@@ -1558,40 +1559,45 @@ public function delete($id)
-                $po->payed_amount -= $previous_amount;
+            // Block deletion if any GRN records reference this PRF
+            if (requested_grn::where('PRF_no', $id)->exists()) {
+                return $this->errorRespose("Cannot delete PRF. One or more GRN records are linked to this PRF.");
+            }
 
-                $po->save();
+            $order_id = $prf->order_id;
+            $order = Order::find($order_id);
+            $po = Purchase_order::where('order_id', $order_id)->first();
+            $po_total_payment = $po->total_payment;
+            $project_id = $order->project_id;
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$project_id'. (undefined)

(UnusedLocalVariable)